src/Paprika/Store/StorageFanOutPage.cs

```diff
@@ -102,6 +102,24 @@ public void Report(IReporter reporter, IPageResolver resolver, int level)
             }
         }
     }
+    public void Destroy(IBatchContext batch, in NibblePath prefix)
+    {
+        // Destroy the Id entry about it
+        Set(prefix, ReadOnlySpan<byte>.Empty, batch);
+
+        // Destroy the account entry
+        // SetAtRoot<FanOutPage>(batch, account, ReadOnlySpan<byte>.Empty, ref Data.StateRoot);
+
+        // Remove the cached
+        batch.IdCache.Remove(prefix.UnsafeAsKeccak);
+        var index = GetIndex(prefix);
+        var addr = Data.Addresses[index];
+        if (addr.IsNull)
+        {
+            // recycleForReuse
+        }
+        TNext.Wrap(batch.GetAt(addr)).Destroy(batch, prefix.SliceFrom(ConsumedNibbles));
```

Comment:
This requires walking through the `Map` with `EnumerateAll` and checking whether there's an entry that matches the prefix. On match, `map.Delete(item)`. This should be on each level can cache things. You could even extend `SlottedArray` with a `Delete(in NibblePath prefix)` that would the scan for you?

Reply:
Added some change. Haven't created a new method for deleting by prefix. Will do it once the logic is correctly implemented. Have a look - @Scooletz